library/std/src/sys/pal/unix/thread.rs

```diff
@@ -142,12 +142,16 @@ impl Thread {
         }
     }
 
-    #[cfg(any(
-        target_os = "freebsd",
-        target_os = "dragonfly",
-        target_os = "openbsd",
-        target_os = "nuttx"
-    ))]
+    #[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
+    pub fn set_name(name: &CStr) {
+        unsafe {
+            let res = libc::pthread_setname_np(libc::pthread_self(), name.as_ptr());
```

Comment:
Does the name need to be truncated as we do with the above `linux` version?

Reply:
Nevermind, I see your above comment.

Reply:
Actually - you could unify this if you wanted.  Change `#[cfg(target_os = "linux")]` to #[cfg(any(target_os = "linux", target_os = "freebsd", target_os = "dragonfly"))]`, make `name` mut with an `allow(unused_mut)`, and then add:

```rust
#[cfg(target_os = "linux")} {
    // Linux limits the allowed length of `name`, other platforms always truncate it.
    const TASK_COMM_LEN: usize = 16;
    let name = truncate_cstr::<{ TASK_COMM_LEN }>(name);
}
```